src/cryptonote_core/pos.cpp

```
@@ -1044,13 +1044,27 @@ round_state goto_wait_for_next_block_and_clear_round_data(round_context &context
   return round_state::wait_for_next_block;
 }
 
+// Returns true if the blockchain's top block differs from the tip cached in
+// the round context — either because a new block arrived (height changed) or
+// because a same-height reorg replaced the top block (hash changed at the
+// same height).
+static bool chain_top_changed(round_context const &context, cryptonote::Blockchain const &blockchain)
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)